src/qweb/extensions.ts

```diff
@@ -67,8 +67,10 @@ export function makeHandlerCode(
     }
   } else {
     // if we get here, then it is an expression
+    // we need to capture every variable in it
     putInCache = false;
-    code = ctx.formatExpression(value);
+    ctx.rootContext.shouldDefineScope = true;
```

Comment:
no need to define scope here, it is done in `captureExpression`

Reply:
done